types/function-bind/ts3.2/function-bind-tests.ts

```diff
@@ -1,88 +0,0 @@
-import bind = require('function-bind/ts3.2');
-
-declare const string: string;
-declare const number: number;
-declare const boolean: boolean;
-
-/**
- * The `expectType` function from https://www.npmjs.com/package/tsd,
- * except instead of returning `void`, it returns `T`.
- */
-declare function expectType<T>(value: T): T;
-
-// $ExpectType (thisArg: any, start?: number | undefined, end?: number | undefined) => any[]
-const slice = expectType<(thisArg: any, start?: number, end?: number) => any[]>(
-    bind.call(Function.call, Array.prototype.slice),
-);
-
-// $ExpectType (start?: number | undefined, end?: number | undefined) => any[]
-expectType<(start?: number, end?: number) => any[]>(bind.call(Function.call, Array.prototype.slice, null));
-
-// $ExpectType (end?: number | undefined) => any[]
-expectType<(end?: number) => any[]>(bind.call(Function.call, Array.prototype.slice, ['a'], 1));
-
-slice(['a']);
-
-// $ExpectType (...args: string[]) => boolean
-expectType<(...args: string[]) => boolean>(bind.call(Boolean, null, String(), '2', '3', '4', '5'));
-
-// $ExpectType (...args: string[]) => boolean
-expectType<(...args: string[]) => boolean>(bind.apply(Boolean, [null, '1', '2', '3', '4', '5']));
```

Comment:
This gets correctly inferred in **TS 3.3+** without needing `expectType<…>`:
<https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b21787307a30f1d7b7d8e80bc58517e3576266bf/types/function-bind/ts3.9/function-bind-tests.ts#L26-L30>